src/watertap_contrib/reflo/core/solar_energy_base.py

```diff
@@ -56,16 +67,90 @@ class SolarEnergyBaseData(UnitModelBlockData):
         ),
     )
     CONFIG.declare(
-        "solar_energy_type",
+        "solar_model_type",
+        ConfigValue(
+            default=SolarModelType.surrogate,
+            domain=In(SolarModelType),
+            description="Solar model type construction flag",
+            doc="""Indicates what type of solar model will be constructed""",
+        ),
+    )
+    CONFIG.declare(
+        "surrogate_model_file",
         ConfigValue(
-            default="PV",
-            domain=In(SolarEnergyType),
-            description="Indicates type of solar energy source",
+            default=None,
+            domain=str,
+            description="Path to surrogate model file",
+            doc="""User provided surrogate model .json file. Must be in same directory as unit model file.""",
+        ),
+    )
+    CONFIG.declare(
+        "dataset_filename",
+        ConfigValue(
+            default=None,
+            domain=str,
+            description="Path to data file",
+            doc="""""",
+        ),
+    )
+    CONFIG.declare(
+        "dataset_bounds",
+        ConfigValue(
+            default=dict(),
+            domain=dict,
+            description="Dict of bounds to use for training dataset",
+            doc="""""",
+        ),
+    )
+    CONFIG.declare(
+        "input_variables",
+        ConfigValue(
+            default=None,
+            domain=dict,
+            description="Dict of names, bounds, and units for surrogate input variables",
+            doc="""""",
+        ),
+    )
+    CONFIG.declare(
+        "output_variables",
+        ConfigValue(
+            default=None,
+            domain=dict,
+            description="Dict of names, bounds, and units for surrogate output variables",
+            doc="""""",
+        ),
+    )
+    CONFIG.declare(
+        "scale_training_data",
+        ConfigValue(
+            default=True,
+            domain=bool,
+            description="Flag to scale surrogate model training data",
+            doc="""""",
+        ),
+    )
+    CONFIG.declare(
+        "number_samples",
+        ConfigValue(
+            default=100,
+            domain=int,
+            description="Number of samples from dataset to build surrogate",
+            doc="""""",
+        ),
+    )
+    CONFIG.declare(
+        "training_fraction",
+        ConfigValue(
+            default=0.8,
+            domain=float,
+            description="Fraction of dataset to use as training data for surrogate",
+            doc="""""",
```

Comment:
Seems highly geared towards more of a `solar_energy_surrogate_base` rather than the possibility for a physical model, but then again, I haven't finished reading all the code yet.

Reply:
I think you are right and is something I realized too late in this implementation... we only have one physical solar model as of now and I believe the only thing this base class is doing for it is creating the ``heat`` and ``electricity`` variables.